cip/1.accepted/CIP2015-05-13-EXISTS.adoc

```diff
@@ -4,102 +4,185 @@
+The semantics of https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RelationshipsPattern[<RelationshipsPattern>] and <SimpleExistentialSubquery> can be defined as syntax transformations to <FullExistentialSubquery>.
 
-==== Subquery exists
+==== <RelationshipsPattern>
 
-When checking for the existence of subqueries, variables and parameters from the outer query are available in the subquery.
-Variables introduced in the subquery are not available on the outside.
+A https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RelationshipsPattern[<RelationshipsPattern>] _RP_ is effectively replaced by the expression
 
-For each matching subgraph evaluated with `EXISTS {}`, the result value must be `true` if the subquery finds at least one matching row.
-If no matches are found, `false` should be returned.
+`EXISTS { MATCH _RP_ RETURN 1 }`
 
-The subquery used in `EXISTS {}` has to follow the following rules:
+==== <SimpleExistentialSubquery>
 
- * It has to be a read-only query. Updating the graph as part of a predicate is not allowed.
- * It cannot end in a `RETURN` clause, nor may it contain `UNION`.
- * When `WITH` ends the subquery, it is able to change the cardinality of the subquery.
- In other words, since the `WITH` clause may include `WHERE`, `SKIP`, and/or `LIMIT`, the clause can with these sub-clauses turn a query that produces rows into one that does not.
- * It may use nested `EXISTS {}` predicates.
+A <SimpleExistentialSubquery> containing a <SimpleMatch> _SM_ is effectively replaced by the expression
 
-==== Simple subquery exists
+`EXISTS { MATCH _SM_ RETURN 1 }`
 
-When the subquery can be described with a single `MATCH-WHERE` clause, the `MATCH` keyword can be omitted, as in example 2A and B.
-The difference between this form and a simple pattern predicate, which is already available, is that this form allows for introducing new variables inside the `EXISTS {}` scope.
+==== <FullExistentialSubquery>
+
+A <FullExistentialSubquery> _FES_ is effectively evaluated as follows:
+
+* Let _OUTER_VARIABLES_ be the current working record for which the expression containing _FES_ is evaluated.
+* Let _NESTED_QUERY_ be the https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RegularQuery[<RegularQuery>] immediately contained in _FES_.
+* Let _RESULT_TABLE_ be the table resulting from evaluating _NESTED_QUERY_ on a driving table comprising _OUTER_VARIABLES_.
+* Case:
+** If _RESULT_TABLE_ is an empty table (cardinality is zero), then the result of _FES_ is [underline]#*_False_*#.
+** Otherwise, the result of _FES_ is [underline]#*_True_*#.
```

Comment:
This explanation is excellent! Just as a detail, perhaps we ought to call out explicitly that all fields of the resulting table are ignored?

Reply:
That GQL spec lingo. You would not say anything else. _RESULT_TABLE_ is a spec variable which implicitly "dies" with the end of the rule block. However, we can add an informative note w.r.t you point.

```suggestion
** Otherwise, the result of _FES_ is [underline]#*_True_*#.

Note that all fields in _RESULT_TABLE_ are ignored and only the number of rows in _RESULT_TABLE_ is relevant for the result of _FES_.
```